### PR TITLE
Added the First circular plot (Polar Cluster) from openair package #5966

### LINKED ISSUE #5966: More circular plots from openair package
There are 6 more circular plots provided by the Openair package.
Percentile Rose
Frequency Rose
Polar Plot
Polar Cluster
Polar Annulus
Polar Frequency
These may become 6 dialogues or (say) 2 dialogues with 3 radio buttons in each or...?  From a quick look I fail to see the connections that would help in structuring the dialogues.  Once decided they should be easy to construct and code, because they have similar options to the Wind/Pollution Rose function.

This is work initially for @HawardKetoyoMsatsi with @Wycklife probably doing the coding then.

### PR DESCRIPTION
**Pull request by HawardKetoyoMsatsi**
Fixes #5966 

@africanmathsinitiative/developers. This is ready for review.

### CODE REVIEW

**Comment by lloyddewit** [comment]
@shadrackkibet please could you review? thanks

File: instat/dlgPolarCluster.vb [hunk 1]
```diff
@@ -1,3 +1,116 @@
...
+
+    Private Sub dlgPolarCluster_Load(sender As Object, e As EventArgs) Handles MyBase.Load
+        If bFirstLoad Then
+            InitialiseDialog()
+            bFirstLoad = False
+        End If
+
+        If bReset Then
+            SetDefaults()
+            SetRcodeForControls(bReset)
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:34]: This should done outside the if statement. Always run when the dialog is loaded.

File: instat/dlgPolarCluster.vb [hunk 2]
```diff
@@ -1,3 +1,116 @@
...
+        End If
+        bReset = False
+        TestOkEnabled()
+
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:39]: Always have a single space after each sub.

File: instat/dlgPolarCluster.vb [hunk 3]
```diff
@@ -1,3 +1,116 @@
...
+
+        ucrReceiverWindDirection.SetParameter(New RParameter("wd", 2))
+        ucrReceiverWindDirection.Selector = ucrSelectorPolarCluster
+        ucrReceiverWindDirection.SetParameterIsString()
+
+        ucrReceiverX.SetParameter(New RParameter("x", 3))
+        ucrReceiverX.Selector = ucrSelectorPolarCluster
+        ucrReceiverX.SetParameterIsString()
+
+        ucrInputColor.SetParameter(New RParameter("cols", 12))
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:60]: why is the parameter position 12?

File: instat/dlgPolarCluster.vb [hunk 4]
```diff
@@ -1,3 +1,116 @@
...
+        dctColor.Add(“Accent”, Chr(34) & "Accent" & Chr(34))
+        dctColor.Add(“Dark2”, Chr(34) & "Dark2" & Chr(34))
+        dctColor.Add(“Paired”, Chr(34) & "Paired" & Chr(34))
+        dctColor.Add(“Pastel1”, Chr(34) & "Pastel1" & Chr(34))
+        dctColor.Add(“Pastel2”, Chr(34) & "Pastel2" & Chr(34))
+        dctColor.Add(“Set1”, Chr(34) & "Set1" & Chr(34))
+        dctColor.Add(“Set2”, Chr(34) & "Set2" & Chr(34))
+        dctColor.Add(“Set3”, Chr(34) & "Set3" & Chr(34))
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:68]: Looks like the quotes i.e `“Accent”/"Accent"` are different here. Did you copy and paste from somewhere?

File: instat/dlgPolarCluster.vb [hunk 5]
```diff
@@ -1,3 +1,116 @@
...
+        ucrInputColor.SetItems(dctColor)
+        ucrInputColor.SetDropDownStyleAsNonEditable()
+
+        ucrNudAngularScale.SetParameter(New RParameter("angle", 14))
+        ucrNudAngularScale.SetMinMax(0, 360)
+
+        ucrNudNoOfClusters.SetParameter(New RParameter("n.clusters", 14))
+        ucrNudNoOfClusters.SetMinMax(1,)
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:76]: Why is the parameter position 14? and same for two parameters?

File: instat/dlgPolarCluster.vb [hunk 6]
```diff
@@ -1,3 +1,116 @@
...
+
+    End Sub
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:78]: space

File: instat/dlgPolarCluster.vb [hunk 7]
```diff
@@ -1,3 +1,116 @@
...
+        ucrSelectorPolarCluster.Reset()
+        ucrReceiverPollutant.SetMeAsReceiver()
+
+        clsPolarClusterFunction.SetPackageName("openair")
+        clsPolarClusterFunction.SetRCommand("polarCluster")
+
+        ucrBase.clsRsyntax.SetBaseRFunction(clsPolarClusterFunction)
+    End Sub
+
+    Private Sub SetRcodeForControls(bReset As Boolean)
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:95]:
```suggestion
    Private Sub SetRCodeForControls(bReset As Boolean)
```

File: instat/dlgPolarCluster.vb [hunk 8]
```diff
@@ -1,3 +1,116 @@
...
+        SetRCode(Me, ucrBase.clsRsyntax.clsBaseFunction, bReset)
+    End Sub
+
+    Private Sub TestOkEnabled()
+        If ucrReceiverPollutant.IsEmpty Then
+            ucrBase.OKEnabled(False)
+        Else
+            ucrBase.OKEnabled(True)
+        End If
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:104]:
```suggestion
            ucrBase.OKEnabled(ucrReceiverPollutant.IsEmpty)
```

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

File: instat/frmMain.resx [hunk 9]
```diff
@@ -4173,6 +4140,36 @@
...
+    <value>mnuBar</value>
+  </data>
+  <data name="&gt;&gt;mnuBar.Type" xml:space="preserve">
+    <value>System.Windows.Forms.MenuStrip, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </data>
+  <data name="&gt;&gt;mnuBar.Parent" xml:space="preserve">
+    <value>$this</value>
+  </data>
+  <data name="&gt;&gt;mnuBar.ZOrder" xml:space="preserve">
+    <value>3</value>
```
Comment (shadrackkibet) [inline instat/frmMain.resx:4171]: Any changes on `frmMain` should be done on a separate PR and merged as soon as possible!

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

File: instat/dlgPolarCluster.vb (hunk 5 above)
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:76]: Is this correct? ` ucrNudNoOfClusters.SetMinMax(1,)`

File: instat/dlgPolarCluster.vb [hunk 10]
```diff
@@ -1,3 +1,116 @@
...
+        ucrNudNoOfClusters.SetParameter(New RParameter("n.clusters", 14))
+        ucrNudNoOfClusters.SetMinMax(1,)
+
+    End Sub
+    Private Sub SetDefaults()
+
+        clsPolarClusterFunction = New RFunction
+
+        clsPolarClusterFunction.AddParameter("cols", Chr(34) & "Accent" & Chr(34), iPosition:=6)
+        clsPolarClusterFunction.AddParameter("angle", "45", iPosition:=14)
```
Comment (shadrackkibet) [inline instat/dlgPolarCluster.vb:84]: check your parameter positions.

**Review by shadrackkibet — CHANGES_REQUESTED** [review]

**Comment by HawardKetoyoMsatsi** [comment]
I am closing this pull request because there are conflicting frmmain resource files.

**Pull request closed without merging**